### PR TITLE
fix: Moq1203 false positives for ReturnsAsync and Callback chaining

### LINKED ISSUE #849: Incorrect detection/flagging of rule `Moq1203` after upgrading to `v0.4.0`
### Description
Existing Setups break after upgrading from `0.3.1` to `0.4.0` due to incorrect flagging of **Moq1203**.

### Steps To Reproduce
``` csharp
public class Moq1203Tests
{
    public interface IFoo
    {
        Task<int> BarAsync();
    }

    [Fact]
    public void Test_Moq1203()
    {
        var moq = new Mock<IFoo>(MockBehavior.Strict);

        // Flags Moq1203.
        moq.Setup(x => x.BarAsync())
            .ReturnsAsync(1);

        // Flags Moq1203.
        moq.Setup(x => x.BarAsync())
            .Callback(() => { })
            .Returns(Task.FromResult(1));

        // OK.
        moq.Setup(x => x.BarAsync())
            .Returns(Task.FromResult(1));

        // OK.
        moq.Setup(x => x.BarAsync())
            .Returns(Task.FromResult(1))
            .Callback(() => { });
    }
}
```

### Environment
- .NET 10
- Moq.Analyzers v0.4.0

### PR DESCRIPTION
**Pull request by rjmurillo**
## Summary
- Walk the full method chain in `HasReturnValueSpecification` instead of only checking the immediate parent, fixing false positives when `Callback()` precedes `Returns()`/`Throws()`
- Recognize `ReturnsAsync` and `ThrowsAsync` as valid return value specification methods alongside `Returns` and `Throws`
- Replace string-based method name matching with symbol-based detection using `MoqKnownSymbols` and `ISymbolExtensions`, consistent with project conventions
- Fix `IsInstanceOf` to handle reduced extension methods (via `ReducedFrom`) so extension methods like `ReturnsAsync`/`ThrowsAsync` are correctly identified
- Add tests exercising the `OverloadResolutionFailure` candidate symbol fallback path
- Consolidate `AnalyzerVerifier` overloads and fix `AllAnalyzersVerifier` reflection lookup
- Simplify `HasReturnValueSymbol` to scan all candidate symbols, not just `OverloadResolutionFailure`

Fixes #849

## Test plan
- [x] All 20 regression tests pass (ReturnsAsync, Callback+Returns, Callback+ReturnsAsync, Callback+Throws)
- [x] All 122 Moq1203 tests pass (including 8 new OverloadResolutionFailure tests)
- [x] Full test suite (1925 tests) passes with no regressions
- [x] All changed lines have coverage; remaining gaps are pre-existing unreachable guard clauses
- [ ] CI passes

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Bug Fixes**
  * Analyzer now reliably detects return/exception specifications (Returns/ReturnsAsync/Throws/ThrowsAsync) across fluent chains (including Callback) and handles async and overload-resolution edge cases.
* **Documentation**
  * Guidance and examples updated to cover async variants, chain-aware placement, and clarified applicability/edge cases.
* **Tests**
  * Added extensive regression and scenario tests for async chains, callback combinations, overload-resolution situations, and custom mock scenarios.

### CODE REVIEW

File: src/Analyzers/MethodSetupShouldSpecifyReturnValueAnalyzer.cs [hunk 1]
```diff
@@ -146,17 +149,48 @@ private static bool HasReturnValueSpecification(IInvocationOperation setupInvoca
...
-            // Check if it's a method call and if the method name is Returns or Throws
-            return symbolInfo.Symbol is IMethodSymbol method &&
-                   (string.Equals(method.Name, "Returns", StringComparison.Ordinal) || string.Equals(method.Name, "Throws", StringComparison.Ordinal));
+            if (HasReturnValueSymbol(symbolInfo))
+            {
+                return true;
+            }
+
+            current = memberAccess.Parent as InvocationExpressionSyntax;
         }
```
[the remark this answers is not in the document]
Comment (rjmurillo) [reply src/Analyzers/MethodSetupShouldSpecifyReturnValueAnalyzer.cs]: Valid observation. Parenthesizing a Setup call (`(mock.Setup(...)).Returns(42)`) would bypass the chain walk because the parent is `ParenthesizedExpressionSyntax` rather than `MemberAccessExpressionSyntax`.

However, this pattern is extremely rare in practice (no occurrences in any Moq test suite or real-world codebase I have found). Addressing it here would be scope creep beyond the issue #849 fix.

I will open a separate issue to track this as a follow-up improvement.

File: tests/Moq.Analyzers.Test/MethodSetupShouldSpecifyReturnValueAnalyzerTests.cs [hunk 2]
```diff
@@ -45,23 +46,150 @@ public static IEnumerable<object[]> TestData()
...
+
+            // Callback before Throws (generic form) should not cause false positive
+            ["""new Mock<IFoo>().Setup(x => x.DoSomething("test")).Callback(() => { }).Throws<InvalidOperationException>();"""],
+
+            // Callback before Throws (instance form) should not cause false positive
+            ["""new Mock<IFoo>().Setup(x => x.DoSomething("test")).Callback(() => { }).Throws(new InvalidOperationException());"""],
+        ];
+
+        return data.WithNamespaces().WithMoqReferenceAssemblyGroups();
+    }
```
[the remark this answers is not in the document]
Comment (rjmurillo) [reply tests/Moq.Analyzers.Test/MethodSetupShouldSpecifyReturnValueAnalyzerTests.cs]: This is the same parenthesized expression edge case discussed in the earlier review thread. We chose to track it as a separate follow-up in #887 rather than address it in this PR, since:
1. The parenthesized Setup pattern is extremely rare in practice
2. Adding the test without the corresponding fix would be a failing test
3. It is unrelated to the issue #849 fix (Callback chaining / ReturnsAsync)

The fix and test will be delivered together in the follow-up issue.
